model/modeldecoder/v2/decoder.go

```diff
@@ -968,6 +974,9 @@ func mapToSpanModel(from *span, event *model.APMEvent) {
 		mapToServiceModel(from.Context.Service, &event.Service)
 		mapToAgentModel(from.Context.Service.Agent, &event.Agent)
 	}
+	if !from.Context.Service.Target.Type.IsSet() {
+		event.Service.Target = inferFromResource(from.Context.Destination.Service.Resource)
```

Comment:
Moving the service destination resource check outside of the function call leaves the code inside that function slightly cleaner, also, we're not immediately escaping that pointer to the heap, although, we are going to anyway in this case.

```suggestion
		if res := from.Context.Destination.Service.Resource; res.IsSet() {
			target := targetFromDestinationResource(res.Val)
			event.Service.Target = &target
		}
```